### PR TITLE
Port OP_SPLATARRAY, OP_NEWRANGE, OP_DEFINED, OP_PUTSTRING, OP_CHECKKEYWORD, OP_CONCATSTRINGS

### PR DESCRIPTION
**Pull request by noahgibbs**

### CODE REVIEW

**Comment by noahgibbs** [comment]
Merging another PR with this one since they're both pretty straightforward (and that avoids more merge conflicts.)

File: yjit.c [hunk 1]
```diff
@@ -135,6 +135,11 @@ get_iseq_flags_has_opt(rb_iseq_t* iseq) {
     return iseq->body->param.flags.has_opt;
 }
 
+int
+get_iseq_body_param_num(rb_iseq_t* iseq) {
```
Comment (maximecb) [inline yjit.c:139]: Will merge regardless, but we might need to prefix all the functions we export with `rb_` or `rb_yjit_` to respect CRuby conventions. @XrXr can you comment?

**Comment by maximecb** [comment]
Very good :)

File: yjit.c (hunk 1 above)
Comment (XrXr) [reply yjit.c:139]: Yea, we should do that. We can probably fix these in one go when we start keeping a fully green CI. The CI check for symbol prefixes is probably yelling already but CI is too noisy at the moment.

Comment (noahgibbs) [reply yjit.c:139]: Yeah, most of CI is currently broken :-/ I've fixed this in the gen_send PR.

**Pull request merged**